python/semantic_kernel/connectors/ai/anthropic/services/anthropic_chat_completion.py

```diff
@@ -254,39 +283,172 @@ def _create_chat_message_content(
-        text_content = ""
-
-        if stream_event.delta and hasattr(stream_event.delta, "text"):
-            text_content = stream_event.delta.text
-
-        items: list[STREAMING_ITEM_TYPES] = [StreamingTextContent(choice_index=content_block_idx, text=text_content)]
-
+        """Create a streaming chat message content object from a content block."""
+        items: list[FunctionCallContent | StreamingTextContent] = []
         finish_reason = None
-        if isinstance(stream_event, RawMessageDeltaEvent):
-            if stream_event.delta.stop_reason:
-                finish_reason = ANTHROPIC_TO_SEMANTIC_KERNEL_FINISH_REASON_MAP[stream_event.delta.stop_reason]
 
-            metadata["usage"].completion_tokens = stream_event.usage.output_tokens
+        if isinstance(stream_event, TextEvent):
+            items.append(StreamingTextContent(choice_index=0, text=stream_event.text))
+        elif isinstance(stream_event, ContentBlockStopEvent):
+            tool_use_block = stream_event.content_block
+            items.append(
+                FunctionCallContent(
+                    id=tool_use_block.id,
+                    index=stream_event.index,
+                    name=tool_use_block.name,
+                    arguments=json.dumps(tool_use_block.input) if tool_use_block.input else None,
+                )
+            )
+        elif isinstance(stream_event, RawMessageDeltaEvent):
+            finish_reason = ANTHROPIC_TO_SEMANTIC_KERNEL_FINISH_REASON_MAP[stream_event.delta.stop_reason]
+            metadata["usage"]["output_tokens"] = stream_event.usage.output_tokens
 
         return StreamingChatMessageContent(
-            choice_index=content_block_idx,
+            choice_index=0,
             inner_content=stream_event,
             ai_model_id=self.ai_model_id,
             metadata=metadata,
-            role=AuthorRole(role) if role else AuthorRole.ASSISTANT,
+            role=AuthorRole.ASSISTANT,
             finish_reason=finish_reason,
             items=items,
         )
+
+    def update_settings_from_function_call_configuration_anthropic(
+        self,
+        function_choice_configuration: FunctionCallChoiceConfiguration,
+        settings: "PromptExecutionSettings",
+        type: "FunctionChoiceType",
+    ) -> None:
+        """Update the settings from a FunctionChoiceConfiguration."""
+        if (
+            function_choice_configuration.available_functions
+            and hasattr(settings, "tools")
+        ):
+            settings.tools = [
+                self.kernel_function_metadata_to_function_call_format_anthropic(f)
+                for f in function_choice_configuration.available_functions
+            ]
+
+            settings.tool_choice = { "type" : type }
+
+            if (
+                settings.function_choice_behavior
+                and settings.function_choice_behavior.type_ == FunctionChoiceType.REQUIRED
+            ):
+                settings.tool_choice = { "type": "any" }
+
```

Comment:
thanks was going to ask about this one. only auto and any can be matched

Reply:
I adjusted it to not setup tools in the case of `NONE` behavior

Reply:
The basic idea of `FunctionChoiceBehavior.NoneInvoke()` is that the model knows the functions, but cannot provide you with a function call. 

So for example if you ask:

`<user> : What is the name of the function to get the actual time?`
`<assistant> : The function to get the current time is 'time-now'.`

So if we try to implement the behavior, we need to set the `tools` even if the Behavior is `NoneInvoke`, so the models knows the tools. Iam still not sure if Anthropic can support this behavior.

I think raising an error instead of skipping the tool information is still the better option.

@moonbox3 What do you think about this ?

Reply:
I agree, @nmoeller.